### ISSUE TITLE
Enhance Yorkie Cluster Functionality for Cluster Reliability

### ISSUE THREAD

**Issue by krapie** [yorkie-team/yorkie#825]
**Description**:

Enhance yorkie cluster functionality for cluster reliability.
We can introduce following K8s features to improve the cluster reliability:
- Pod [Resource Request/Limit](https://kubernetes.io/ko/docs/concepts/configuration/manage-resources-containers/)
- Pod [HPA(Horizontal Pod Autoscaling)](https://kubernetes.io/ko/docs/tasks/run-application/horizontal-pod-autoscale/)
- Pod [PodDisruptionBudget](https://kubernetes.io/ko/docs/concepts/workloads/pods/disruptions/#pdb-example)

For this, it will be good to load test single yorkie pod.

**Why**:

Currently yorkie cluster only supports minimal requirements for the cluster.
We need to add features to safely manage yorkie pods for a cluster reliability.

**Comment by taeng0204** [comment]
Hi, I'm interested in this issue. Can I try it?

**Comment by krapie** [comment]
@taeng0204 Sure!

**Comment by taeng0204** [comment]
@krapie When working on this issue, should the cluster be based on EKS? Or is minikube okay?

**Comment by krapie** [comment]
@taeng0204 The cluster will most likely to be based on the EKS cluster, considering our use case of current Yorkie SaaS API running on the EKS cluster.

**Comment by window9u** [comment]
Hello  @taeng0204 😊
With the 2025 OSSCA project now active, we're looking to keep issues moving. If you're still actively working on this one, please let us know.
If not, we'll unassign you soon to open it up for others who are interested. Thanks for your understanding and contributions to Yorkie!

**Comment by esc-beep** [comment]
Hello, Can I take this one?

**Comment by kokodak** [comment]
@esc-beep Yes, you can take it! If you have any questions, feel free to let me know.
